soh/src/overlays/actors/ovl_En_Heishi3/z_en_heishi3.c

```diff
@@ -206,6 +206,9 @@ void func_80A55D00(EnHeishi3* this, PlayState* play) {
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play) &&
         (this->respawnFlag == 0)) {
         Flags_SetEventChkInf(EVENTCHKINF_CAUGHT_BY_CASTLE_GUARDS);
+        if (LINK_IS_ADULT) {
+            gPlayState->linkAgeOnLoad ^= 1; // Test for EVENT
```

Comment:
Could this be moved to a hook for scene load using the entrance below? Or does the age need to be child before triggering the scene change?

Reply:
I think there's a few things in this PR that touch source that shouldn't. This possibly being one of them but you do have to ensure Link loads as a child in that scene prior otherwise the game will crash.